onmt/translate/Translator.lua

```diff
@@ -281,6 +284,21 @@ function Translator:buildTargetWords(pred, src, attn)
     end
   end
 
+  if self.args.replace_unk_tagged then
```

Comment:
Could you merge this chunk with the loop above? Something like:

```lua
if self.args.replace_unk or self.args.replace_unk_tagged then
  for i = 1, #tokens do
    if tokens[i] == onmt.Constants.UNK_WORD then
      local _, maxIndex = attn[i]:max(1)
      local source = src[maxIndex[1]]

      if self.phraseTable and self.phraseTable:contains(source) then
        tokens[i] = self.phraseTable:lookup(source)
      elseif self.args.replace_unk then
        tokens[i] = source
      elseif self.args.replace_unk_tagged then
        tokens[i] = '｟unk:' .. source .. '｠'
      end
    end
  end
end
```

Reply:
I went for the naive way because I have literally never written more than 3 lines of lua, but I’ll try my hand at more than naively copy pasting and editing :)